### ISSUE TITLE
Any loop keyword directly after label causes string index out of bounds

### ISSUE THREAD

**Issue by parker0phil** [CodeNarc/CodeNarc#85]
> Using Codenarc plugin from Gradle 2.2.1

The node.col value seems to carry over the length of the label.

Also applies if the keyword is 'for'.

Example is in a spock test:

```
when:
def foo = true
while (foo){
```
works fine, but:
```
when:
while (true){
```
causes:
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 10
	at java.lang.String.substring(String.java:1907)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.getAt(DefaultGroovyMethods.java:3243)
	at org.codehaus.groovy.runtime.dgm$238.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:270)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:52)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:124)
	at org.codenarc.rule.formatting.AbstractSingleSpaceAfterKeywordAstVisitor.checkForSingleSpaceAndOpeningParenthesis(AbstractSingleSpaceAfterKeywordAstVisitor.groovy:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:266)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:51)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:44)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.codenarc.rule.formatting.SpaceAfterWhileAstVisitor.visitWhileLoop(SpaceAfterWhileRule.groovy:37)
	at org.codehaus.groovy.ast.stmt.WhileStatement.visit(WhileStatement.java:39)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:35)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:165)
	at org.codenarc.rule.ClassCodeVisitorSupportHack.visitBlockStatement(ClassCodeVisitorSupportHack.java:95)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:51)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:101)
	at org.codenarc.rule.ClassCodeVisitorSupportHack.visitClassCodeContainer(ClassCodeVisitorSupportHack.java:45)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:112)
	at org.codenarc.rule.ClassCodeVisitorSupportHack.visitConstructorOrMethod(ClassCodeVisitorSupportHack.java:55)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:123)
	at org.codenarc.rule.ClassCodeVisitorSupportHack.visitMethod(ClassCodeVisitorSupportHack.java:65)
	at org.codenarc.rule.AbstractAstVisitor.visitMethod(AbstractAstVisitor.java:170)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1039)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:50)
	at org.codenarc.rule.ClassCodeVisitorSupportHack.visitClass(ClassCodeVisitorSupportHack.java:20)
	at org.codenarc.rule.AbstractAstVisitor.visitClass(AbstractAstVisitor.java:148)
	at org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
	at org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:141)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:40)
	at org.codenarc.rule.Rule$applyTo.call(Unknown Source)
	at org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
	at org.codenarc.ant.AntFileSetSourceAnalyzer.processFile(AntFileSetSourceAnalyzer.java:168)
	at org.codenarc.ant.AntFileSetSourceAnalyzer.access$000(AntFileSetSourceAnalyzer.java:41)
	at org.codenarc.ant.AntFileSetSourceAnalyzer$1.run(AntFileSetSourceAnalyzer.java:157)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:722)
```

**Comment by rozza** [comment]
Possibly related I'm also seeing this causing an issue:

   for (i in 1..8) {

but not for:

   def range = 1..8
   for (i in range) {

**Comment by rozza** [comment]
A couple more examples failing in spock:

    when:
    def iter = [1, 2, 3].iterator()

    then:
    while(iter.hasNext()) {

But when both declared in the same `then` block its ok:

    then:
    def iter = [1, 2, 3].iterator() 
    while(iter.hasNext()) {

**Comment by chrismair** [comment]
For what it's worth, I reproduced these errors using Groovy 1.7.5, and verified that they do not occur when running with Groovy 2.1.9. 

We are considering changing the minimum required Groovy version from 1.7 to something higher for the next release, anyway, so that may be how we resolve this issue.

**Comment by rozza** [comment]
Unfortunately, they are reproducible when running Groovy 2.3.6

**Comment by chrismair** [comment]
We have tests for these examples and have tested with Groovy 2.3.6 and do not see the issue with CodeNarc 0.23. Please let us know if this problem still occurs with CodeNarc 0.23.

**Comment by chrismair** [comment]
Closing since this seems to be resolved with CodeNarc 0.23. If that is not the case, please reopen.
